saleor/graphql/discount/mutations/promotion/promotion_create.py

```diff
@@ -200,6 +203,27 @@ def _save_m2m(
             rule.channels.set(channels)
 
     @classmethod
-    def send_sale_notifications(cls, manager, instance):
-        # TODO: implement the notifications
-        pass
+    def send_promotion_notifications(
+        cls, manager: "PluginsManager", instance: models.Promotion
+    ):
+        cls.call_event(manager.promotion_created, instance)
+        cls.send_promotion_toggle_notification(manager, instance)
+
+    @staticmethod
+    def send_promotion_toggle_notification(
+        manager: "PluginsManager", instance: models.Promotion
+    ):
+        """Send a notify about starting or ending promotion if it hasn't been sent yet.
+
+        Send the notification when the start date is before the current date and the
+        promotion is not already finished.
+        """
+        now = datetime.now(pytz.utc)
+
+        start_date = instance.start_date
+        end_date = instance.end_date
+
+        if (start_date and start_date <= now) and (not end_date or not end_date <= now):
+            manager.promotion_toggle(instance)
+            instance.notification_sent_at = now
```

Comment:
IMHO This could not be true. Currently, you're tracking the time when you pass the notification task to the celery. But the notification can be sent after 1 sec, or 10 minutes or 1 hour.

Reply:
That's true. I can add a double check in webhook.

Reply:
On another side, I'm not sure if it needs double check. If it doesn't send it that moment, it will be send a later by a sheduler.